src/app/components/content/admin/add-partner/add-partner.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthenticateService } from '../../../../services/authenticate.service';
+import { User } from '../../../../models/User';
+
+@Component({
+  selector: 'app-add-partner',
+  templateUrl: './add-partner.component.html',
+  styleUrls: ['./add-partner.component.scss']
+})
+export class AddPartnerComponent implements OnInit {
+
+  /** Invalid form values bools */
+  emailIsInvalid = false;
+  passwordIsInvalid = false;
+  confirmPasswordIsInvalid = false;
+  companyNameIsInvalid = false;
+  gotNetworkError = false;
```

Comment:
I seem to understand these booleans are used to display a message under each input field when that input is still invalid? You could also use the forms own error determination method `form.controls[controlName].hasError('required'|'email'|'min')` or `form.controls[controlName].valid` to check its validity.

Reply:
I know did it like this so if you needed to expand it you wouldn't be needed to change the code

Reply:
that is correct, but if you do expand the form, you'd need to add 3 more lines, including an extra boolean property, just to check if the input field was filled in correctly, while Angular provides that feature out of the box.